src/Components/Star/StarRating/StarRating.js

```diff
@@ -0,0 +1,197 @@
+        headers: { 'Content-type': 'application/json; charset=UTF-8' },
+      }).then(() => {
+        this.props.callApi();
+      });
+
+      this.setState({
+        isHover: false,
+        hoverRateValue: [false, false, false, false, false],
+        rateValue: prevRateValue,
+      });
+
+      return;
+    }
+
+    if (isClickedStarActive) {
+      prevRateValue.fill(false, 0, clickedIndex + 1);
+
+      const rating = prevRateValue.filter(value => value).length;
+
+      fetch('http://localhost:1313/rating/7', {
+        method: 'PATCH',
+        body: JSON.stringify({
+          rating: rating,
+        }),
+        headers: { 'Content-type': 'application/json; charset=UTF-8' },
+      }).then(() => {
+        this.props.callApi();
+      });
+
+      this.setState({
+        isHover: false,
+        hoverRateValue: [false, false, false, false, false],
+        rateValue: prevRateValue,
+      });
+
+      return;
+    }
+
+    if (!isClickedStarActive) {
+      prevRateValue.fill(true, 0, clickedIndex + 1);
+
+      const rating = prevRateValue.filter(value => value).length;
+
+      fetch('http://localhost:1313/rating/7', {
+        method: 'PATCH',
+        body: JSON.stringify({
+          rating: rating,
+        }),
+        headers: { 'Content-type': 'application/json; charset=UTF-8' },
+      }).then(() => {
+        this.props.callApi();
+      });
+
+      this.setState({
+        isHover: false,
+        hoverRateValue: [false, false, false, false, false],
+        rateValue: prevRateValue,
+      });
+
+      return;
+    }
+  };
+
+  handleStarMousehover = hoveredIndex => {
+    // console.log('MOUSE hover');
+    const prevRateValue = [...this.state.hoverRateValue];
+    const isClickedStarActive = prevRateValue[hoveredIndex];
+    const isNextStarActive = prevRateValue[hoveredIndex + 1];
+
+    if (isClickedStarActive && isNextStarActive) {
+      prevRateValue.fill(false, hoveredIndex + 1);
+
+      this.setState({
+        isHover: true,
+        hoverRateValue: prevRateValue,
+      });
+
+      return;
+    }
+
+    if (isClickedStarActive) {
+      prevRateValue.fill(false, 0, hoveredIndex + 1);
+
+      this.setState({
+        isHover: true,
+        hoverRateValue: prevRateValue,
+      });
+
+      return;
+    }
+
+    if (!isClickedStarActive) {
+      prevRateValue.fill(true, 0, hoveredIndex + 1);
+
+      this.setState({
+        isHover: true,
+        hoverRateValue: prevRateValue,
+      });
+
+      return;
+    }
+  };
+
+  handleStarMouseout = () => {
+    // console.log('MOUSE OUT');
+
+    this.setState({
+      isHover: false,
+      hoverRateValue: [false, false, false, false, false],
+    });
+  };
+
+  checkIsActive = star => {
+    if (this.state.isHover) {
+      if (this.state.hoverRateValue[star]) {
+        return 'activeStar';
+      }
+
+      return 'inactiveStar';
+    }
+
+    if (this.state.rateValue[star]) {
+      return 'activeStar';
+    }
+
+    return 'inactiveStar';
```

Comment:
- 결국엔 경우의 수가 2개이기 때문에, 이렇게 너무 많은 조건을 사용할 필요가 없어 보입니다! 패턴 하나 소개해드립니다! 참고해주세요!
```suggestion
    const { isHover, hoverRateValue, rateValue } = this.state;
    const isStarActive = (isHover && hoverRateValue[star]) || (rateValue[star]);
    return isStarActive ? 'activeStar' : 'inactiveStar';
```
- 조건이 복잡할 경우엔 조건을 변수에 담고 해당 변수의 이름을 조건의 내용을 알 수 있는 것으로 네이밍 해주는 것을 통해, 조건을 직접 해석하지 않아도 조건이 어떤 의미를 가지는지 파악할 수 있도록 해주시는 것이 좋습니다.

Reply:
성훈님 제가 작성한 코드에서는 isHover와 hoverRateValue[star]가 둘 다 false면 그대로 return 하면서 끝나는데
참고로 올려주신 코드에서 isHover와 hoverRateValue[star] 둘 다 false여도 rateValue[star]가 true인 경우가 나와서요.

제가 && ||을 이용해서 고쳐보려고 했는데 도저히 떠오르지가 않아서

    if (isHover) {
      return hoverRateValue[star] ? 'activeStar' : 'inactiveStar';
    }

    return isStarActive ? 'activeStar' : 'inactiveStar';

이렇게까지만 작성했습니다... ㅠㅠ;  더 좋은 방법이 있을까요?

Reply:
아뇨-! 얼리 리턴을 사용해서 작성해주신 현재 코드 깔끔한 것 같습니다! LGTM ✨